src/django/api/tests.py

```diff
@@ -3378,3 +3378,25 @@ def test_creates_record(self):
         log = DownloadLog.objects.first()
         self.assertEqual(expected_path, log.path)
         self.assertEqual(expected_record_count, log.record_count)
+
+
+class TilePermissionsTest(APITestCase):
+    def setUp(self):
+        self.tile_path = reverse('tile', kwargs={
+            'layer': 'facilitygrid',
+            'cachekey': '1567700347-1-95f951f7',
+            'z': 6, 'x': 15, 'y': 29,
+            'ext': 'pbf',
+        })
+
+    @override_settings(ALLOWED_HOSTS=['testserver', '.allowed.org'])
+    def test_allowed_hosts_can_fetch_tiles(self):
```

Comment:
Consider using the `@override_switch` decorator so this test can assert a 200 response.

```diff
diff --git a/src/django/api/tests.py b/src/django/api/tests.py
index 7b928ef..2a79c9a 100644
--- a/src/django/api/tests.py
+++ b/src/django/api/tests.py
@@ -3390,12 +3390,11 @@ class TilePermissionsTest(APITestCase):
         })
 
     @override_settings(ALLOWED_HOSTS=['testserver', '.allowed.org'])
+    @override_switch('vector_tile', active=True)
     def test_allowed_hosts_can_fetch_tiles(self):
         response = self.client.get(self.tile_path, {},
                                    HTTP_REFERER='http://allowed.org/')
-        # Expect a 404 instead of 200 because the test database does not
-        # have vector tiles loaded
-        self.assertEqual(404, response.status_code)
+        self.assertEqual(200, response.status_code)
 
     def test_disallowed_hosts_cannot_fetch_tiles(self):
         response = self.client.get(self.tile_path)

```

Reply:
Thanks, this is exactly what was missing!